tarpc/src/transport/mod.rs

```diff
@@ -0,0 +1,84 @@
+use std::io::{self, Read, Write};
+use std::time::Duration;
+
+/// A factory for creating a listener on a given address.
```

Comment:
The word address only comes up in the comments, and I think it's confusing. In a vacuum, it doesn't seem obvious to me what the connection is between an address and a transport, or what you even mean by address.

Some ideas
- Remove the word address 
- Rename the trait to `Address`
- Provide more detailed comments about what addresses have to do with this trait. Maybe an example of what address would mean for tcp, which might make it more concrete/understandable.